Formula/hledger.rb

```diff
@@ -15,25 +28,19 @@ class Hledger < Formula
     sha256 "22f6bd98c3b593aa5ff55deea85b2c7aa4da9e4d075f1eed05cbca71aefed0ab" => :high_sierra
   end
 
-  depends_on "cabal-install" => :build
-  depends_on "ghc" => :build
+  depends_on "haskell-stack" => :build
+  # Use brew's ghc, assuming it's compatible, instead of letting stack fetch it. Any faster ?
+  # Seems to be failing, maybe from ghc/snapshot minor version mismatch.
+  # depends_ on "ghc" => :build
 
   uses_from_macos "ncurses"
   uses_from_macos "zlib"
 
-  resource "hledger-web" do
-    url "https://hackage.haskell.org/package/hledger-web-1.16.2/hledger-web-1.16.2.tar.gz"
-    sha256 "daa4b68d8491d0a5716ee2ac39520a31bef6a1ae6b36ddc0f531b81616c237ce"
-  end
-
-  resource "hledger-ui" do
-    url "https://hackage.haskell.org/package/hledger-ui-1.16.2/hledger-ui-1.16.2.tar.gz"
-    sha256 "90f69e1e2f9fa66c535b7f61144b3fb681f1629f313c89d423b4569722224faf"
-  end
-
   def install
-    install_cabal_package "hledger", "hledger-web", "hledger-ui", "brick-0.50.1", :using => ["happy", "alex"]
-    # help cabal pick a good brick version, https://github.com/Homebrew/homebrew-core/pull/49010#issuecomment-574719702
+    # similar to the stack install command at https://hledger.org/download
+    system "stack", "update"
+    system "stack", "install", "--local-bin-path=#{bin}", "--resolver=lts-15",
```

Comment:
There is no way to tell stack to use a file? Because this bypasses all the checksum restrictions we have.

Reply:
Thanks for the review @SMillerDev. As you probably know this is an experimental formula exploring solutions for the problem discussed at https://github.com/Homebrew/brew/pull/6945.

Yes I believe I could make it use the checksummed tarballs for the hledger packages. All other haskell dependencies will still be fetched from stackage though. Homebrew's cabal-based formulae do this too, except those are less repeatable than fetching from stackage.